### PR TITLE
Implement Network Rule Resource

### PR DESCRIPTION
**Pull request by NavisJ**
# Description
Implement network rule resource

# ISSUE TYPE
- Feature Pull Request

##### RESOURCE OR DATASOURCE NAME
Network rule

##### OUTPUT

image
image

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 80% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Unit tests
- [x] Acceptance tests

### CODE REVIEW

**Comment by NavisJ** [comment]
run e2e test

File: examples/resources/powerscale_network_rule/resource.tf [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+# Available actions: Create, Update, Delete and Import
+# After `terraform apply` of this example file for the first time, you will create a network rule on the PowerScale
+
+# PowerScale subnet allows you to manage the network rule on the Powerscale array
+resource "powerscale_networkpool" "pool_test" {
```
Comment (taohe1012) [inline examples/resources/powerscale_network_rule/resource.tf:22]: here should be powerscale_network_rule

File: examples/resources/powerscale_network_rule/resource.tf [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+# Available actions: Create, Update, Delete and Import
+# After `terraform apply` of this example file for the first time, you will create a network rule on the PowerScale
+
+# PowerScale subnet allows you to manage the network rule on the Powerscale array
```
Comment (RayLiu7) [inline examples/resources/powerscale_network_rule/resource.tf:21]: network rule?

**Comment by NavisJ** [comment]
run e2e test

File: examples/resources/powerscale_network_rule/resource.tf (hunk 1 above)
Comment (NavisJ) [reply examples/resources/powerscale_network_rule/resource.tf]: Modified. Thank you!

File: examples/resources/powerscale_network_rule/resource.tf (hunk 2 above)
Comment (NavisJ) [reply examples/resources/powerscale_network_rule/resource.tf]: Modified. Thank you!

**Review by taohe1012 — APPROVED** [review]

**Review by RayLiu7 — APPROVED** [review]

**Review by baoy1 — APPROVED** [review]
Need to rebase the latest develop.

**Comment by NavisJ** [comment]
run e2e test

**Comment by NavisJ** [comment]
> Need to rebase the latest develop.

Rebased.

**Comment by NavisJ** [comment]
run e2e test

**Pull request merged**